### DISCUSSION TITLE
Simple question: How to upgrade?

### DISCUSSION THREAD

**Discussion by osseonews** [pocketbase/pocketbase#550] [Q&A]
Sorry, simple question. If we are running Pocketbase on DO, how do we upgrade to the latest version? Do we first stop the server, and then run `wget` again to pull the latest version, and then start it up again. Any worries about the database getting changed?

**Comment by ganigeorgiev** [comment]
For upgrade you can just upload/replace the binary with the new one and restart the systemd/process (eg. `systemctl restart pocketbase.service`).

At least on linux, you don't need to first stop the server when removing the old binary, because the os will keep the old inode until there is a process using it, so it is safe to call `rm`, `mv`, `unlink`, etc. before stopping the server. `rsync` is also safe (unless you use the `--in-place` flag):

```sh
# replace the pocketbase binary
rsync -avz -e ssh /local/path/to/pocketbase root@YOUR_DROPLET_IP:/root/pb/pocketbase

# restart pocketbase
systemctl restart pocketbase.service
```

> Any worries about the database getting changed?

PocketBase DB migrations are embedded in the binary and they will be auto applied on `pocketbase serve` (aka. on systemd restart).

**It is always a good idea to have a backup of your `pb_data` before upgrading PocketBase.** To backup remote->local, you could just reverse the position of the `rsync` arguments:

```sh
rsync -avz -e ssh root@YOUR_DROPLET_IP:/root/pb/pb_data/ /local/path/to/pb_data/
```

---

_Please keep in mind that PocketBase is still under active development and until we reach v1.0.0 full backward compatiblity is not guaranteed and therefore make sure to check the release notes that comes with each new version before upgrading - https://github.com/pocketbase/pocketbase/releases._
